File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        candidate, count = -1, 0

    # Find a candidate
        for num in nums:
           if count == 0:
             candidate = num
           count += (1 if num == candidate else -1)
 
    # Validate the candidate
        if nums.count(candidate) > len(nums) // 2:
           return candidate
        return -1    
        

                    

        

File: 0169-majority-element/0169-majority-element.dart
class Solution {
  int majorityElement(List<int> nums) {
        
      Map<int, int> counterMap = {};

      int result = 0;

      for(int n in nums) {
        counterMap[n] = (counterMap[n] ?? 0) + 1;
      }

      counterMap.forEach((key, value) {
         if(value > nums.length / 2) {
            result = key;
         }
      });
    return result;

  }
}